### DISCUSSION TITLE
Flash/highlight new unread messages until hotkey pressed.

### DISCUSSION THREAD

**Discussion by jitspoe** [Chatterino/chatterino2#3676] [Ideas]
While streaming, I'm really bad about accidentally missing questions / inadvertently ignoring chat, so I'd love a feature that:

- Flashes a highlight on a new line of text.
- New chat line stays highlighted until a hotkey is pressed.
- (Hotkey can be pressed from outside of the application to avoid the need to alt-tab out of game or whatever I'm working on.)
- After the hotkey is pressed, the next line of chat is highlighted, and so on, until all messages are read.

**Comment by LosFarmosCTL** [comment]
> (Hotkey can be pressed from outside of the application to avoid the need to alt-tab out of game or whatever I'm working on.)

This will most likely not work sadly, see #3673

Aside from that, basically what you want is kind of a selector where you can highlight some message and then move that selection up/down so that you always know what message you read last?

**Reply by jitspoe** [reply]
> This will most likely not work sadly, see #3673

Replied on that thread.

> Aside from that, basically what you want is kind of a selector where you can highlight some message and then move that selection up/down so that you always know what message you read last?

Yeah, wouldn't really have to be that fancy. I was just thinking of utilizing something like the highlight color that's already there, then, after you press a key, it would switch that line of chat back to unhighlighted (or highlight in whatever color it would normally do as per the settings) and highlight the next line of chat.

Ideally it wouldn't scroll the new line of chat off screen as well if there's a lot of new chat.

**Reply by LosFarmosCTL** [reply]
I think if someone implements it looking into making it a bit more general could be really great, I frequently wish that I could do anything with messages using hotkeys instead of having to grab my mouse, some system for selecting them would be amazing for that.

But, considering how especially highlights are implemented right now, this would probably be a pretty major change and require some extensive rework of a lot of current mechanics.

It would be a really nice addition though and imo definitely something to keep in mind for the future.

**Reply by pajlada** [reply]
I think a combination of global hotkeys, using the "Draw a line below the most recent message before switching applications." option, and maybe some more styling options to the option would be an alright solution to this problem.

That way, you would have a bar of some kind as an indicator for messages you haven't read/responded to yet, and a global hotkey would be able to move the line to the bottom or hide it until new messages come in (as if you focused the application)
